src/main/java/org/apache/datasketches/DoublesSortedView.java

```diff
@@ -36,39 +38,75 @@ public interface DoublesSortedView extends SortedView {
    * the CDF of the input data stream. Therefore, each point represents the fractional density of the distribution
-   * between from zero. For example, if one of the returned values is 0.5, then the splitPoint corresponding to that
-   * value would be the median of the distribution.</p>
+   * from zero to the given point. For example, if one of the returned values is 0.5, then the splitPoint corresponding
+   * to that value would be the median of the distribution.</p>
+   *
+   * @param splitPoints the given array of quantile values or splitPoints. This is a sorted, monotonic array of unique
+   * values in the range of (minValue, maxValue). This array should not include either the minValue or the maxValue.
+   * The returned array will have one extra interval representing the very top of the distribution.
+   * @param searchCrit if INCLUSIVE, each interval within the distribution will include its top value and exclude its
+   * bottom value. Otherwise, it will be the reverse.  The only exception is that the top portion will always include
+   * the top value retained by the sketch.
+   * @return an array of points that correspond to the given splitPoints, and represents the input data distribution
+   * as a CDF.
+   */
+  default double[] getCDF(double[] splitPoints, QuantileSearchCriteria searchCrit) {
+    checkDoublesSplitPointsOrder(splitPoints);
+    final int len = splitPoints.length + 1;
+    final double[] buckets = new double[len];
+    for (int i = 0; i < len - 1; i++) {
+      buckets[i] = getRank(splitPoints[i], searchCrit);
+    }
+    buckets[len - 1] = 1.0;
+    return buckets;
+  }
+
+  /**
+   * Returns an array of values where each value is a number in the range [0.0, 1.0].
+   * The size of this array is one larger than the size of the input splitPoints array.
    *
-   * <p>If <i>isCdf</i> is false, the points in the returned array are not monotonic and represent the discrete
-   * derivative of the CDF, or the Probability Mass Function (PMF). Each returned point represents the fractional
+   * <p>The points in the returned array are not monotonic and represent the discrete derivative of the CDF,
+   * which is also called the Probability Mass Function (PMF). Each returned point represents the fractional
    * area of the total distribution which lies between the previous point (or zero) and the given point, which
-   * corresponds to the given splitPoint.<p>
+   * corresponds to the given splitPoint.</p>
    *
-   * @param splitPoints the given array of quantile values or splitPoints. This is a sorted, unique, monotonic array
-   * of values in the range of (minValue, maxValue). This array should not include either the minValue or the maxValue.
+   * @param splitPoints the given array of quantile values or splitPoints. This is a sorted, monotonic array of unique
+   * values in the range of (minValue, maxValue). This array should not include either the minValue or the maxValue.
```

Comment:
still value here. and I am not sure about "should not include". including them is not very useful, but we don't enforce anything like that

Reply:
Changed it to "does not need to" here and in FloatsSortedView and GenericSortedView.

Reply:
And for both getPMF() and getCDF()